web/src/components/tasks/CannabisPriorityStatus.tsx

```diff
@@ -0,0 +1,342 @@
+
+  const [displayMWPriorityType, setDisplayMWPriorityType] = useState(false);
+  const [displayVeteranPriorityType, setDisplayVeteranPriorityType] = useState(false);
+  const [displayImpactZonePriorityType, setDisplayImpactZonePriorityType] = useState(false);
+  const [displaySocialEquityPriorityType, setDisplaySocialEquityPriorityType] = useState(false);
+  const [displayNoPriorityType, setDisplayNoPriorityType] = useState(false);
+
+  const priorityTypes = useMemo(() => {
+    return [
+      ...priorityTypesObj.minorityOrWomen,
+      ...priorityTypesObj.veteran,
+      ...priorityTypesObj.impactZone,
+      ...priorityTypesObj.socialEquity,
+    ];
+  }, []);
+
+  useEffect(() => {
+    if (userData && userData.taskItemChecklist && displayFirstTab) {
+      const priorityTypeSelected = priorityTypes.some((key) => userData.taskItemChecklist[key] === true);
+
+      if (priorityTypeSelected || userData.taskItemChecklist[priorityTypesObj.noneOfTheAbove]) {
+        setDisplayNextTabButton(true);
+      } else setDisplayNextTabButton(false);
+
+      if (priorityTypeSelected && userData.taskItemChecklist[priorityTypesObj.noneOfTheAbove]) {
+        update({
+          ...userData,
+          taskItemChecklist: {
+            ...userData.taskItemChecklist,
+            [priorityTypesObj.noneOfTheAbove]: false,
+          },
+        });
+      }
+    }
+
+    if (userData && userData.taskItemChecklist && !displayFirstTab) {
+      const minorityOrWomenPriorityTypeSelected = priorityTypesObj.minorityOrWomen.some((key) => {
+        if (key in userData.taskItemChecklist) return userData.taskItemChecklist[key] === true;
+        else return false;
+      });
+      const veteranPriorityTypeSelected = priorityTypesObj.veteran.some((key) => {
+        if (key in userData.taskItemChecklist) return userData.taskItemChecklist[key] === true;
+        else return false;
+      });
+
+      const impactZonePriorityTypeSelected = priorityTypesObj.impactZone.some((key) => {
+        if (key in userData.taskItemChecklist) return userData.taskItemChecklist[key] === true;
+        else return false;
+      });
+
+      const socialEquityPriorityTypeSelected = priorityTypesObj.socialEquity.some((key) => {
+        if (key in userData.taskItemChecklist) return userData.taskItemChecklist[key] === true;
+        else return false;
+      });
+
+      setDisplayMWPriorityType(minorityOrWomenPriorityTypeSelected ? true : false);
+      setDisplayVeteranPriorityType(veteranPriorityTypeSelected ? true : false);
+      setDisplayImpactZonePriorityType(impactZonePriorityTypeSelected ? true : false);
+      setDisplaySocialEquityPriorityType(socialEquityPriorityTypeSelected ? true : false);
+      setDisplayNoPriorityType(userData.taskItemChecklist[priorityTypesObj.noneOfTheAbove] ? true : false);
+    }
+  }, [userData, update, priorityTypes, displayFirstTab]);
+
+  const handleCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
might rename this to "handleNoneOfTheAboveCheckboxChange" or something because there's a lot of other checkboxes on the page that this does not handle